From 007c00a5f52c4910f810c986d3eba43af8369496 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 16 Mar 2016 15:46:27 +0200
Subject: [PATCH 0955/1240] icu: ppv2: fix bad icu indices for PPv2 interrupt

interrupts 19 and 20 had the same icu id as interrup 18
which caused an override of in the ICU configuration

Change-Id: Ia7de633069fa1cd7ed1ac3f26fe9f5b46c84de07
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28298
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/misc/mvebu_icu.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/mvebu_icu.c b/drivers/misc/mvebu_icu.c
index 3498494..4134fe6 100644
--- a/drivers/misc/mvebu_icu.c
+++ b/drivers/misc/mvebu_icu.c
@@ -86,6 +86,7 @@ static struct icu_irq irq_map_ns_multi[NS_MULTI_IRQS] = {
 	{36, 3, 0}, /* PPv2     AXI monitor */
 
 	{38,  4, 0}, /* PPv2 Misc */
+
 	{39,  5, 0}, /* PPv2 irq */
 	{40,  6, 0}, /* PPv2 irq */
 	{41,  7, 0}, /* PPv2 irq */
@@ -100,8 +101,8 @@ static struct icu_irq irq_map_ns_multi[NS_MULTI_IRQS] = {
 	{50, 16, 0}, /* PPv2 irq */
 	{51, 17, 0}, /* PPv2 irq */
 	{52, 18, 0}, /* PPv2 irq */
-	{52, 19, 0}, /* PPv2 irq */
-	{52, 20, 0}, /* PPv2 irq */
+	{53, 19, 0}, /* PPv2 irq */
+	{54, 20, 0}, /* PPv2 irq */
 
 	{78, 21, 0}, /* MG irq */
 	{88, 22, 0}, /* EIP-197 ring-0 */
-- 
1.9.1

